modules/foreman_host.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Bernhard Hopfenmüller (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_host
+short_description: Manage Foreman hosts
+description:
+  - "Manage Foreman host Entities"
+  - "This beta version can create, power on/off, delete hosts from preexisting host groups"
+  - "Uses https://github.com/SatelliteQE/nailgun"
+version_added: "2.7"
+author:
+  - "Bernhard Hopfenmueller (@Fobhep) ATIX AG"
+requirements:
+  - "nailgun >= 0.29.0"
+  - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    required: false
+    default: true
+    type: bool
+
+# NEEDS REWORK
+
+
+'''
+
+EXAMPLES = '''
+- name: "Create a host"
+  foreman_host:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_host"
+    hostgroup: my_hostgroup
+    state: present
+
+- name: "Switch off an existing host"
+  foreman_host:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_host"
+    enabled: false
+    state: present
```

Comment:
`present` means powered-off? Should we add aliases here?

Reply:
no it does not :)
I plan to add a designated parameter for the power state.
And since a host can clearly be present but switched off, I would not remove the "classic" state parameter

Reply:
I could think of `on` and `off` as part of the `state` of the host, while `present` does not touch the power state. This would reduce some logic around the fact, that `absent` cannot be combined with a `power_state` of `on`.

Reply:
mmh sounds actually like an idea - I will try to build something like that